recipes/install.rb

```diff
@@ -0,0 +1,90 @@
+# encoding: UTF-8
+# Cookbook Name: apache_zookeeper
+# Recipe:: install
+
+
+
+# Download remote file first. If we fail here, no need to continue
+version_tag   = "zookeeper-#{node['apache_zookeeper']['version']}"
+download_url  = ::File.join(node['apache_zookeeper']['mirror'], 
+                            "#{version_tag}/#{version_tag}.tar.gz")
+download_path = ::File.join(Chef::Config[:file_cache_path], 
+                            "#{version_tag}.tar.gz")
+# This cannot be download from a mirror
+verify_url    = ::File.join(node['apache_zookeeper']['dist'], 
+                            "dist/zookeeper",
+                            "#{version_tag}/#{version_tag}.tar.gz.sha1")
+
+remote_file "#{download_path}.sha1" do
+  action :create_if_missing 
+  source verify_url
+  backup false
+end
+
+# TODO: Use PGP signature instead of sha1 for verification
+remote_file download_path do
+  action :create_if_missing
+  source download_url
+  backup false
+  #verify { 
+  #  `sha1sum -c #{download_path}.sha1`
```

Comment:
It seems like you didn't get this to work. Do you mind removing this and sha1 remote file?